vignettes/s3-vector.Rmd

```diff
@@ -1147,6 +1148,45 @@ vec_cast.pizza_percent.double <- function(x, to, ...) percent(x)
 vec_cast.double.pizza_percent <- function(x, to, ...) vec_data(x)
 ```
 
+### Arithmetic
+
+Writing double dispatch methods for `vec_arith()` is currently more awkward than writing them for `vec_ptype2()` or `vec_cast()`. We plan to improve this in the future. For now, you can use the following instructions.
+
+If you define a new type and want to write `vec_arith()` methods for it, you'll need to provide a new single dispatch S3 generic for it of the following form:
+
+```{r, eval=FALSE}
+#' @export
+#' @method vec_arith my_type
+vec_arith.my_type <- function(op, x, y, ...) {
+  UseMethod("vec_arith.my_type", y)
+}
+```
+
+Note that this actually functions as both an S3 method for `vec_arith()` and an S3 generic called `vec_arith.my_type()` that dispatches off `y`. roxygen2 only recognizes it as an S3 generic, so you have to register the S3 method part of this with an explicit `@method` call.
+
+After that, you can define double dispatch methods, but you still need an explicit `@method` tag to ensure it is registered with the correct generic:
+
+```{r, eval=FALSE}
+#' @export
```

Comment:
Do these methods need to be exported?

Reply:
Yes, otherwise it removes things like this:

<img width="620" alt="Screen Shot 2022-09-19 at 10 08 33 AM" src="https://user-images.githubusercontent.com/19150088/191037136-469ac9b9-b583-441c-a0c4-b07e4c848d5f.png">

And then you get this error:

```r
> vctrs::vec_arith("+", duration_days(5), Sys.Date())
Error in UseMethod("vec_arith.clock_duration", y) : 
  no applicable method for 'vec_arith.clock_duration' applied to an object of class "Date"
```